File: web/src/shared/types/verification.ts
/**
 * Reason for why verification cannot be completed
 */
export type VerificationReason =
  | 'busy'
  | 'away'
  | 'cannot_find'
  | 'wrong_location'
  | 'not_an_issue';

/**
 * Type of verification being performed
 */
export type VerificationType = 'existence' | 'fix';

/**
 * Result of a verification
 */
export type VerificationResult =
  | 'confirmed'
  | 'not_found'
  | 'not_fixed'
  | 'cannot_verify';

/**
 * Status of a verification request
 */
export type VerificationStatus = 'pending' | 'completed' | 'expired';

/**
 * Issue verification record
 */
export interface IssueVerification {
  id: string;
  issueId: string;
  verificationType: VerificationType;
  assignedTo?: string;
  assignedToName?: string;
  verifiedBy?: string;
  verifiedByName?: string;
  result?: VerificationResult;
  reason?: VerificationReason;
  note?: string;
  photoId?: string;
  requestedAt: string;
  respondedAt?: string;
  status: VerificationStatus;
}

/**
 * Request to create a verification request for an issue
 */
export interface RequestVerificationRequest {
  type: VerificationType;
  assignTo?: string;
  message?: string;
}

/**
 * Response from verification history endpoint
 */
export interface VerificationHistoryResponse {
  items: IssueVerification[];
}

/**
 * Map verification reasons to i18n label keys
 */
export const VERIFICATION_REASON_LABELS: Record<VerificationReason, string> = {
  busy: 'verification.reasons.busy',
  away: 'verification.reasons.away',
  cannot_find: 'verification.reasons.cannotFind',
  wrong_location: 'verification.reasons.wrongLocation',
  not_an_issue: 'verification.reasons.notAnIssue',
};

/**
 * Map verification results to i18n label keys
 */
export const VERIFICATION_RESULT_LABELS: Record<VerificationResult, string> = {
  confirmed: 'verification.results.confirmed',
  not_found: 'verification.results.notFound',
  not_fixed: 'verification.results.notFixed',
  cannot_verify: 'verification.results.cannotVerify',
};

/**
 * Verification reasons as array for dropdowns
 */
export const VERIFICATION_REASONS: VerificationReason[] = [
  'busy',
  'away',
  'cannot_find',
  'wrong_location',
  'not_an_issue',
];

File: mobile/lib/shared/models/verification.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'geo_point.dart';

part 'verification.freezed.dart';
part 'verification.g.dart';

/// Reasons why a Ground Admin cannot verify an issue
enum VerificationReason {
  busy,
  away,
  @JsonValue('cannot_find')
  cannotFind,
  @JsonValue('wrong_location')
  wrongLocation,
  @JsonValue('not_an_issue')
  notAnIssue;

  String get displayName => switch (this) {
        busy => 'I am busy right now',
        away => 'I am away from the area',
        cannotFind => 'Cannot find the issue',
        wrongLocation => 'Location seems incorrect',
        notAnIssue => 'This is not actually an issue',
      };
}

/// Verification result types
enum VerificationResult {
  confirmed,
  @JsonValue('not_found')
  notFound,
  @JsonValue('not_fixed')
  notFixed,
  @JsonValue('cannot_verify')
  cannotVerify;

  String get displayName => switch (this) {
        confirmed => 'Confirmed',
        notFound => 'Not Found',
        notFixed => 'Not Fixed',
        cannotVerify => 'Cannot Verify',
      };
}

/// Verification type
enum VerificationType {
  existence,
  fix;

  String get displayName => switch (this) {
        existence => 'Existence Verification',
        fix => 'Fix Verification',
      };
}

/// A pending verification request for a Ground Admin
@freezed
abstract class PendingVerification with _$PendingVerification {
  const PendingVerification._();

  const factory PendingVerification({
    required String verificationId,
    required String issueId,
    required String issueType,
    required String verificationType,
    required GeoPoint location,
    required DateTime requestedAt,
    double? distance,
  }) = _PendingVerification;

  /// Whether this is an existence verification
  bool get isExistenceVerification => verificationType == 'existence';

  /// Whether this is a fix verification
  bool get isFixVerification => verificationType == 'fix';

  /// Formatted distance if available
  String? get formattedDistance {
    if (distance == null) return null;
    if (distance! < 1000) {
      return '${distance!.toInt()}m away';
    }
    return '${(distance! / 1000).toStringAsFixed(1)}km away';
  }

  factory PendingVerification.fromJson(Map<String, dynamic> json) =>
      _$PendingVerificationFromJson(json);
}

/// Response for submitting a verification
@freezed
abstract class VerificationSubmitRequest with _$VerificationSubmitRequest {
  const factory VerificationSubmitRequest({
    required String verificationId,
    required String result,
    String? reason,
    String? note,
    String? photoId,
  }) = _VerificationSubmitRequest;

  factory VerificationSubmitRequest.fromJson(Map<String, dynamic> json) =>
      _$VerificationSubmitRequestFromJson(json);
}

/// Issue verification record
@freezed
abstract class IssueVerification with _$IssueVerification {
  const factory IssueVerification({
    required String id,
    required String issueId,
    required String verificationType,
    String? assignedTo,
    String? verifiedBy,
    String? result,
    String? reason,
    String? note,
    String? photoId,
    required DateTime requestedAt,
    DateTime? respondedAt,
    required String status,
  }) = _IssueVerification;

  factory IssueVerification.fromJson(Map<String, dynamic> json) =>
      _$IssueVerificationFromJson(json);
}

/// List of pending verifications response
@freezed
abstract class PendingVerificationsResponse with _$PendingVerificationsResponse {
  const factory PendingVerificationsResponse({
    required List<PendingVerification> items,
  }) = _PendingVerificationsResponse;

  factory PendingVerificationsResponse.fromJson(Map<String, dynamic> json) =>
      _$PendingVerificationsResponseFromJson(json);
}
